Libs/Core/ctkUtils.cpp

```diff
@@ -395,6 +396,22 @@ bool ctk::copyDirRecursively(const QString &srcPath, const QString &dstPath, boo
   return true;
 }
 
+//-----------------------------------------------------------------------------
+bool ctk::copyFile(const QString& srcPath, const QString& dstPath)
+{
+    std::ifstream src(srcPath.toStdString(), std::ios::binary);
```

Comment:
Stream-based file copy can be 2x slower than dedicated platform-specific file copy (see https://stackoverflow.com/questions/10195343/copy-a-file-in-a-sane-safe-and-efficient-way). The difference can be much more dramatic if you copy remote files (stream-based implementation actually has to download the file to the local computer and then send it to the remote computer). Have you tested the performance on Windows/macOS/linux, with local->local, local->remote, remote->remote file copy? Have you checked what API QFile::copy uses internally? If it already uses some slow copy operation then it is not a regression, but if it used a fast platform-specific API and now we switch to slow API then this is a major regression.

This only works on Windows if the application uses UTF-8 everywhere (the application code page is set to UTF-8), while QFile::copy works well regardless of encoding. Not a deal-breaker, but at least it must be documented.

QFile::copy works well when disk gets full (you can tell from the filename that it is an incomplete file). Not a deal-breaker, but it is a nice feature that we lose by making this change.

Overall, these disadvantages might be acceptable (except if the performance drop is significant) if there is a good motivation to change this, but the very small convenience (avoid adding a somewhat heuristic file name check) in a very special use case might not worth the trouble.